pennylane/tape/qubit_param_shift.py

```diff
@@ -379,3 +379,22 @@ def processing_fn(results):
             return np.apply_along_axis(dot, 0, results)
 
         return tapes, processing_fn
+
+    @property
+    def specs(self):
+
+        if "grad_method" not in self._par_info[0]:
+            self._update_gradient_info()
+
+        # Initialize with the forward pass execution
+        num_executions = 1
+        # Loop over all variables
+        for _, info in self._par_info.items():
+
+            if info["grad_method"] == "A":
+                num_executions += len(info["op"].get_parameter_shift(info["p_idx"]))
+
+        info = super().specs
+        info["num_parameter_shift_executions"] = num_executions
+
+        return info

```

Comment:
I just realised `specs` doesn't work so well with circuits that include templates:
```python
import pennylane as qml
from pennylane import numpy as np

x = np.array([0.1, 0.2])

with qml.tape.QubitParamShiftTape() as tape:
    qml.templates.StronglyEntanglingLayers(np.random.random((3, 2, 3)), wires=range(2))

tape.specs
```
Gives
```
{'by_size': defaultdict(int, {2: 1}),
 'by_name': defaultdict(int, {'StronglyEntanglingLayers': 1}),
 'total_operations': 1,
 'total_observables': 0,
 'num_tape_wires': 2,
 'depth': 1,
 'num_trainable_params': 1,
 'num_parameter_shift_executions': 1}
```
Though in practice, there are many trainable params and there will be many param shift executions :thinking:.
There may not be an easy fix for this though. And also things work nicely on the QNode level as expansion will have happened.

Reply:
I agree, maybe something we can think about as we improve the `Operation` class. (I believe this subtlety is due to moving templates to operations, since they'd be automatically expanded before?)